pathos/multiprocessing.py

```diff
@@ -96,27 +96,27 @@ def __init__(self, *args, **kwds):
             kwds['ncpus'] = kwds.pop('nodes')
         elif arglen:
             kwds['ncpus'] = args[0]
-        self.__nodes = kwds.get('ncpus', cpu_count())
+        self.__nodes = kwds.pop('ncpus', cpu_count())
 
         # Create an identifier for the pool
-        self._id = kwds.get('id', None) #'pool'
+        self._id = kwds.pop('id', None) #'pool'
         if self._id is None:
             self._id = self.__nodes
 
         # Create a new server if one isn't already initialized
-        self._serve()
+        self._serve(**kwds)
         return
     if AbstractWorkerPool.__init__.__doc__: __init__.__doc__ = AbstractWorkerPool.__init__.__doc__ + __init__.__doc__
    #def __exit__(self, *args):
    #    self._clear()
    #    return
-    def _serve(self, nodes=None): #XXX: should be STATE method; use id
+    def _serve(self, nodes=None, **kwds): #XXX: should be STATE method; use id
         """Create a new server if one isn't already initialized"""
         if nodes is None: nodes = self.__nodes
         _pool = __STATE.get(self._id, None)
         if not _pool or nodes != _pool.__nodes:
```

Comment:
needs to also check `_pool._maxtasksperchild`, `_pool.initargs`, `_pool.initializer`

Reply:
```suggestion
        if (
            _pool is None
            or nodes != _pool.__nodes
            or kwds.get('maxtasksperchild') != _pool._maxtasksperchild
            or kwds.get('initargs') != _pool.initargs
            or kwds.get('initializer') != _pool.initializer
       ):
```
like this?

Reply:
it's missing the leading underscores in the latter two.

Reply:
👍 edited the comment

Reply:
I believe you'll also need: (1) corresponding changes to `_clear`, and (2) when `_serve` is called by one of the `map` functions and `kwds={}`, then it does the expected thing by pulling the kwds from the existing `pool`.

Reply:
Added 6c5b537 which should take care of only add a new pool to state if kwds changed, and only clear a pool from state if kwds match. The (2) part I didn't fully get: pool kwds are e.g. `initializer`, whereas map kwds are e.g. `chunksize`. I think it should be OK with this last commit, as the is no overlap between those two kinds of kwds?

Reply:
correct, there is no overlap between map and pool kwds.